volplugin/handlers.go

```diff
@@ -260,8 +260,16 @@ func (dc *DaemonConfig) mount(w http.ResponseWriter, r *http.Request) {
 	}
 
 	volName := volConfig.String()
+	mountCount:= dc.increaseMount(volName)
 
-	if dc.increaseMount(volName) > 1 {
+	if !volConfig.Unlocked && mountCount > 1 {
+		if mountCount > 2 { // keeping counter=2 during locked mounts helps docker unmount from removing the actual mount
```

Comment:
I don't think we want to do this decrease here. If the mount is requested multiple times in parallel this will inflate the value, but still cause docker to issue an unmount and eventually go below 0 (which deliberately causes a panic). Let's somehow make sure this is tested directly if you feel the need to keep it.